pkg/volume/rbd/rbd_util.go

```diff
@@ -96,24 +108,98 @@ func getDevFromImageAndPool(pool, image string) (string, bool) {
 	return "", false
 }
 
-// stat a path, if not exists, retry maxRetries times
-func waitForPath(pool, image string, maxRetries int) (string, bool) {
+// Locate any existing rbd-nbd process mapping given a <pool, image>.
+// Recent versions of rbd-nbd tool can correctly provide this info using list-mapped
+// but older versions of list-mapped don't.
+// The implementation below peeks at the command line of nbd bound processes
+// to figure out any mapped images.
+func getNbdDevFromImageAndPool(pool string, image string) (string, bool) {
+	// nbd module exports the pid of serving process in sysfs
+	basePath := "/sys/block/nbd"
+	// Do not change imgPath format - some tools like rbd-nbd are strict about it.
+	imgPath := fmt.Sprintf("%s/%s", pool, image)
+	for i := 0; ; i++ {
```

Comment:
is this searching through all the pid namespace ? how often will this be called ?

Reply:
Here we are iterating only over nbd devices, inside `/sys/block/nbd`.  This pattern is the same as the existing rbd/krbd method, enumerating over the (k)rbd devices in sysfs.

This method is called with the same frequency as existing getRbdDev calls.